helix-term/src/ui/menu.rs

```diff
@@ -231,8 +231,14 @@ impl<T: Item + 'static> Component for Menu<T> {
             key!(Enter) => {
                 if let Some(selection) = self.selection() {
                     (self.callback_fn)(cx.editor, Some(selection), MenuEvent::Validate);
+                    return close_fn;
+                } else {
+                    return EventResult::Ignored(Some(Box::new(
```

Comment:
Let's change the close_fn definition to be Some(Box(..)) (without the Event Result wrapper), then you can reuse it for both branches